extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java

```diff
@@ -471,135 +480,134 @@ private String generateInvoker(ScheduledBusinessMethodItem scheduledMethod, Clas
         for (Type i : method.parameterTypes()) {
             sigBuilder.append(i.name().toString());
         }
-        String generatedName = DotNames.internalPackageNameWithTrailingSlash(implClazz.name()) + baseName
+        String packageName = DotNames.packageName(implClazz.name());
+        String generatedName = (!packageName.isEmpty() ? packageName + "." : "") + baseName
                 + INVOKER_SUFFIX + "_" + method.name() + "_"
                 + HashUtil.sha1(sigBuilder.toString());
 
         boolean isSuspendMethod = KotlinUtil.isSuspendMethod(method);
 
-        ClassCreator invokerCreator = ClassCreator.builder().classOutput(classOutput).className(generatedName)
-                .superClass(isSuspendMethod ? SchedulerDotNames.ABSTRACT_COROUTINE_INVOKER.toString()
-                        : DefaultInvoker.class.getName())
-                .build();
-
-        MethodCreator invoke;
-        if (isSuspendMethod) {
-            invoke = invokerCreator
-                    .getMethodCreator("invokeBean", Object.class.getName(),
-                            ScheduledExecution.class.getName(), SchedulerDotNames.CONTINUATION.toString());
-        } else {
-            // The descriptor is: CompletionStage invoke(ScheduledExecution execution)
-            invoke = invokerCreator.getMethodCreator("invokeBean", CompletionStage.class, ScheduledExecution.class);
-        }
-
-        // Use a try-catch block and return failed future if an exception is thrown
-        TryBlock tryBlock = invoke.tryBlock();
-        CatchBlockCreator catchBlock = tryBlock.addCatch(Throwable.class);
-        if (isSuspendMethod) {
-            catchBlock.throwException(catchBlock.getCaughtException());
-        } else {
-            catchBlock.returnValue(catchBlock.invokeStaticMethod(
-                    MethodDescriptor.ofMethod(CompletableFuture.class, "failedStage", CompletionStage.class, Throwable.class),
-                    catchBlock.getCaughtException()));
-        }
-
-        String returnTypeStr = DescriptorUtils.typeToString(method.returnType());
-        ResultHandle res;
-        if (isStatic) {
-            if (implClazz.isInterface()) {
-                if (method.parameterTypes().isEmpty()) {
-                    res = tryBlock.invokeStaticInterfaceMethod(
-                            MethodDescriptor.ofMethod(implClazz.name().toString(), method.name(), returnTypeStr));
-                } else {
-                    res = tryBlock.invokeStaticInterfaceMethod(
-                            MethodDescriptor.ofMethod(implClazz.name().toString(), method.name(), returnTypeStr,
-                                    ScheduledExecution.class),
-                            tryBlock.getMethodParam(0));
-                }
+        gizmo.class_(generatedName, cc -> {
+            if (isSuspendMethod) {
+                cc.extends_(ClassDesc.of(SchedulerDotNames.ABSTRACT_COROUTINE_INVOKER.toString()));
```

Comment:
You can also use `Jandex2Gizmo.classDescOf(DotName)`. If you statically import the method (which I would recommend and personally always do), it looks like `classDescOf(SchedulerDotNames.ABSTRACT_COROUTINE_INVOKER)`.